server/public/model/user.go

```diff
@@ -1007,3 +1011,36 @@ type UsersWithGroupsAndCount struct {
 	Users []*UserWithGroups `json:"users"`
 	Count int64             `json:"total_count"`
 }
+
+type UserPostStats struct {
+	LastLogin    int64  `json:"last_login_at,omitempty"`
+	LastStatusAt *int64 `json:"last_status_at,omitempty"`
+	LastPostDate *int64 `json:"last_post_date,omitempty"`
+	DaysActive   *int   `json:"days_active,omitempty"`
+	TotalPosts   *int   `json:"total_posts,omitempty"`
+}
+
+type UserReportQuery struct {
+	User
+	UserPostStats
+}
+
+type UserReport struct {
+	Id          string `json:"id"`
+	Username    string `json:"username"`
+	Email       string `json:"email"`
+	CreateAt    int64  `json:"create_at,omitempty"`
+	DisplayName string `json:"display_name"`
+	UserPostStats
+}
+
+func (u *UserReportQuery) ToReport() UserReport {
+	return UserReport{
+		Id:            u.Id,
+		Username:      u.Username,
+		Email:         u.Email,
+		CreateAt:      u.CreateAt,
+		DisplayName:   u.GetDisplayName(ShowNicknameFullName),
+		UserPostStats: u.UserPostStats,
```

Comment:
Since `UserPostStats` has reference types, they won't be deep copied. I don't know if thats something you were trying to avoid though.

Reply:
Right, no I wasn't I expect them to be copied in. This is what I get for working in JS for too long :P

Reply:
Okay, so it actually does seem to work:
![image](https://github.com/mattermost/mattermost/assets/52460000/c3f1ee93-69ad-4564-bcb9-f27d3c2402c7)

This is the data I expect. Does it work differently with struct composition?

Reply:
Normally it should also contain `UserPostStats` as it was embedded. Sorry, I thought it was a different field. In any case, once you copy pointer values, it will copy the pointers. Which means if you change those values in the clone, it will also get updated in the original struct. But I guess there is no concern here about that.